### PR TITLE
Tests for Search use case + Changes to variable types in entities 

### LINKED ISSUE #42: Search and API Output Test
Create tests for search use case and different potential API outputs

### PR DESCRIPTION
**Pull request by RedFoxity**
Changed dependency on "CommonFridge" to "Fridge" in multiple classes causing input type errors.

Created 6 tests for Search Use Case
Also added docstrings...
1. Search with 1 Ingredient in fridge
2. Search with multiple Ingredients in fridge
3. Search when fridge is empty 4.Search with illegal characters in ingredient input
5. Search with ridiculous input
6. Search with duplicate ingredients

### CODE REVIEW

**Review by EricKarpovits — CHANGES_REQUESTED** [review]
Your Tests aren't actually asserting anything, please make sure that you Assert something instead of printing it out into the console otherwise it will always evaluate to True. You can even evaluate to a hard-coded string but the console is not the best practice.

File: src/main/java/entities/fridge/CommonFridge.java [hunk 1]
```diff
@@ -13,13 +12,13 @@ public class CommonFridge implements Fridge, Serializable {
      * User can make changes to the ingredients list by adding or removing an ingredient.
      * Existing user can set their Ingredient to their previous ingredients list.
      */
-    private ArrayList<CommonIngredient> ingredients;
+    private ArrayList<Ingredient> ingredients;
```
Comment (EricKarpovits) [inline src/main/java/entities/fridge/CommonFridge.java]: Make sure that the rest of the team is okay with these changes from CommonIngredient to Ingredient.

File: src/test/java/tutorial/SearchTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+// Checks how the API handles a Fridge with a single ingredient object in it as input
+    @Test
+    public void SearchWithSingleIngredients() {
+
+        BobFridge.addIngredient(ingFact.create("Carrot"));
+        SearchRequestModel request = new SearchRequestModel("Bob","Lunch/Dinner","300-800 cal", "Less than 30 minutes");
+        SearchResponseModel response = interactor.search(request);
+        System.out.println(response.getRecipes());
```
Comment (EricKarpovits) [inline src/test/java/tutorial/SearchTest.java:65]: Printing out into the console will always evaluate to True.

**Review by EricKarpovits — CHANGES_REQUESTED** [review]

**Comment by EricKarpovits** [comment]
Also make sure to resolve the merge conflicts, otherwise great tests.

File: src/main/java/entities/fridge/CommonFridge.java (hunk 1 above)
Comment (Shaffaan) [reply src/main/java/entities/fridge/CommonFridge.java]: What is the reason for changing all of them from common ingredient to their interfaces?

**Pull request closed without merging**